build-python.sh

```diff
@@ -308,12 +308,16 @@ if test "${INSTALL}" = true; then
         # use regular-old setuptools for these builds, to avoid
         # trying to recompile the shared library
         sed -i.bak -e '/start:build-system/,/end:build-system/d' pyproject.toml
-        echo '[build-system]' >> ./pyproject.toml
-        echo 'requires = ["setuptools"]' >> ./pyproject.toml
-        echo 'build-backend = "setuptools.build_meta"' >> ./pyproject.toml
-        echo "" >> ./pyproject.toml
-        echo "recursive-include lightgbm *.dll *.dylib *.so" > ./MANIFEST.in
-        echo "" >> ./MANIFEST.in
+        {
+            echo '[build-system]'; \
+            echo 'requires = ["setuptools"]'; \
+            echo 'build-backend = "setuptools.build_meta"'; \
+            echo ""; \
+        }  >> ./pyproject.toml
```

Comment:
what is the purpose of these `{}`? Can you please share the exact `shellcheck` warnings this is addressing?

I don't understand the purpose of doing this in a sub-shell, I strongly suspect there's a better way to address whatever warning this is about.

Reply:
The subsequence commands piping into the same file where raising this warning https://www.shellcheck.net/wiki/SC2129

I did it with a subshell as suggested, joining all strings with `\n` and single echo should be fine too

Reply:
Thanks for that explanation. I really find this command-group style difficult to read. 

As it says there

> *This is mainly a stylistic issue, and can freely be ignored.*

So please revert all of the changes like this and add `# shellcheck disable=SC2129` comments to suppress these.